bootstrap/kubeadm/config/manager/manager.yaml

```diff
@@ -20,14 +20,16 @@ spec:
         - /manager
         args:
         - "--leader-elect"
-        - "--metrics-bind-addr=localhost:8080"
```

Comment:
This change should probably not be merged as-is. I needed to remove this from `args` to be able to set this argument through `entrypoint.extend()` in the tiltfile.

Does anyone have an idea how we can override the argument in Tilt without removing this default from all the `manager.yaml` files.?

Reply:
I think we can just append `--metrics-bind-addr=0.0.0.0:8080` to the `entrypoint` here https://github.com/kubernetes-sigs/cluster-api/blob/7c437015e2fdac31d2b422dcf03ad34fff39b5cc/Tiltfile#L242

This would lead to issues if a provider doesn't support `--metrics-bind-addr`, but as far as I'm aware, all providers implement this flag (and if they don't we can always make it configurable in a follow-up PR)

Reply:
Just realized you're already doing this and the args are overwritting the entrypoint. Maybe we can "just" append it to the args instead?

Reply:
I wonder if we should just move all this business logic in tilt-prepare/main.go (problem here is how to hand over the configuration to tilt-prepare)

Reply:
Maybe it would be the easiest to just move "append metrics bind addr" and the addition of the containerPort to tilt-prepare/main.go. We area already doing similar things there.
(@fabriziopandini WDYT?)

Reply:
> Just realized you're already doing this and the args are overwritting the entrypoint. Maybe we can "just" append it to the args instead? Not sure how easy that would be.

I tried just adding it to the `entrypoint`, but is was being overwritten by the `args` from the yaml manifest. Appending to the `args` is not supported by tilt, only overwriting them, but that means we loose the `--leader-elect` and `--feature-gates` flags (or need to duplicate them in Tilt).

> Maybe it would be the easiest to just move "append metrics bind addr" and the addition of the containerPort to tilt-prepare/main.go.

That sounds like a good idea, I will try this next.

Reply:
> Appending to the args is not supported by tilt, only overwriting them, but that means we loose the --leader-elect and --feature-gates flags (or need to duplicate them in Tilt).

Good to know. 

I think mid-term we should probably move more of the logic into the Go binary if feasible. I think the only problem with that is that we have to pass configuration into the binary. 

But for the metrics bind addr it sounds reasonable to me to just implement it in Go as we don't need any configuration. It's just not ideal that we have some of that logic in the Tiltfile and some in tilt-prepare (but as mentioned we should probably resolve that at some point, by moving more into tilt-prepare)